### PR TITLE
[CIAPP-7523] Mark last retry from datadog-ci gate

### PR DESCRIPTION
**Pull request by liashenko**
### What and why?
Mark last retry from datadog-ci gate to be able to send failed gate executions to EvP from the service.

We want to provide better visibility in Datadog in the following cases:
* one or more rule evaluations failed during the last request from datadog-ci
* datadog-ci timed out before finishing the evaluation

Currently Quality Gates service is not aware of internal datadog-ci retries, so we can’t send CIGATE events to EvP on the last request for the cases mentioned above.

More context in the [RFC](https://docs.google.com/document/d/1Cj5i5rs_S85AueP7m60gg4iWYVZ-Opr-SB2VpHzvEvE/edit)

### How?
When it's a last retry attempt or the remaining wait time is 0, add `isLastRetry:true` to the API request.

### Review checklist

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration)

### CODE REVIEW

File: src/commands/gate/evaluate.ts
Comment (romainkomorndatadog) [inline src/commands/gate/evaluate.ts:186]: Not that we should ever get to `attempt === this.maxRetries + 2`, but would this be more appropriate?
```
attempt > this.maxRetries
```

**Review by romainkomorndatadog — APPROVED** [review]
I think this looks good but I'm feeling a little bit shy about approving since I'm not all that familiar with this repo.

Maybe get a second opinion. :)

**Review by andrea-mosk — APPROVED** [review]
LGTM if you have tested it

File: src/commands/gate/evaluate.ts
Comment (liashenko) [reply src/commands/gate/evaluate.ts:186]: it shouldn't happen ever, if it happens datadog-ci retry is broken.

but if it does, we expect to get a single `lastRetry` in the service (because we'll send the event to EvP, and we want to have 1 event per gate execution), so I'd keep it as is if you don't mind

**Pull request merged**
